Documentation/design-docs/hw-intrinsics.md

```diff
@@ -12,47 +12,89 @@ There is a design document for the Arm64 intrinsics: https://github.com/dotnet/c
 
 The reference assemblies for the hardware intrinsics live in corefx, but all of the implementation is in the coreclr repo:
 
-* The C# implementation lives in coreclr/src/shared/System/Runtime/Intrinsics. These are little more than skeleton methods that usually invoke themselves recursively (see Indirect invocation support below).
+* The C# implementation lives in coreclr/src/System.Private.CoreLib/shared/System/Runtime/Intrinsics. These are little more than skeleton methods that are only compiled if needed for indirect invocation.
+
+  * Note that they also exist in the dotnet/corefx repo in src\Common\src\CoreLib\System\Runtime\Intrinsic. These are mirrored between the two repositories.
 
 ## C# Implementation
 
-The hardware intrinsics operate on and produce both primitive types (`int`, `float`, etc.) as well as vector types. The vector types are considered platform-agnostic, though not all platforms define intrinsics on all of these types. They are:
+The hardware intrinsics operate on and produce both primitive types (`int`, `float`, etc.) as well as vector types.
+
+### Platform-agnostic vector types
+
+The vector types supported by one or more target ISAs are supported across platforms, though they extent to which operations on them are available and accelerated is dependent on the target ISA. These are:
 
 * `Vector64<T>` - A 64-bit vector of type `T`. For example, a `Vector64<int>` would hold two 32-bit integers.
+  * Note that `Vector64<T>` intrinsics are supported only on Arm64, and these are not supported for `double`. Support could be added for this, but would require additional handling.
```

Comment:
Should this be "currently" supported?

Is it also worth calling out that `Vector256<T>` is currently only supported by x86 (and would require the SVE extensions to work on ARM)?